### ISSUE TITLE
 I can't publish packages, error: ref HEAD is not a symbolic ref

### ISSUE THREAD

**Issue by Vasiliy-Lavrov** [guigrpa/oao#62]
Hello, I can't publish packages with command:
`oao publish -i "examples/*" --no-master --no-changelog --no-git-commit --no-check-uncommitted --no-confirm --new-version 0.1.0-0205151508.alpha`
 
My workflow:
- I work in my fork
- I create a pull request to the original repository
- My CI (server for continuous integration) can't publish a new alpha version on my private npm repository ( I am using Verdaccio: https://github.com/verdaccio/verdaccio )
 
I have an error with a message: `git ERROR | fatal: ref HEAD is not a symbolic ref`
```
2018-02-05T15:15:15.687Z                  git ERROR | fatal: ref HEAD is not a symbolic ref
2018-02-05T15:15:15.690Z                  git INFO  | 
2018-02-05T15:15:15.692Z                 main ERROR Command 'git symbolic-ref --short HEAD' failed [128] at '.'
2018-02-05T15:15:15.692Z                 main TRACE └── Run cmd git symbolic-ref --short HEAD [CLOSED]

{ Error: Command 'git symbolic-ref --short HEAD' failed [128] at '.'
    at execError (/var/lib/jenkins/workspace/xxx_-_alpha_xxx_PR-100-XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/node_modules/oao/lib/utils/shell.js:193:13)
    at _callee2$ (/var/lib/jenkins/workspace/xxx_-_alpha_xxx_PR-100-XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/node_modules/oao/lib/utils/shell.js:173:20)
    at tryCatch (/var/lib/jenkins/workspace/xxx_-_alpha_xxx_PR-100-XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/node_modules/babel-polyfill/node_modules/regenerator-runtime/runtime.js:65:40)
    at Generator.invoke [as _invoke] (/var/lib/jenkins/workspace/xxx_-_alpha_xxx_PR-100-XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/node_modules/babel-polyfill/node_modules/regenerator-runtime/runtime.js:303:22)
    at Generator.prototype.(anonymous function) [as throw] (/var/lib/jenkins/workspace/xxx_-_alpha_xxx_PR-100-XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/node_modules/babel-polyfill/node_modules/regenerator-runtime/runtime.js:117:21)
    at step (/var/lib/jenkins/workspace/xxx_-_alpha_xxx_PR-100-XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/node_modules/oao/lib/utils/shell.js:28:191)
    at /var/lib/jenkins/workspace/xxx_-_alpha_xxx_PR-100-XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX/node_modules/oao/lib/utils/shell.js:28:402
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:160:7)
  code: 128,
  stdout: '',
  stderr: 'fatal: ref HEAD is not a symbolic ref\n' }
```

This bug is here: https://github.com/guigrpa/oao/blob/master/src/publish.js#L169
 
P.S. GIT doesn't have a branch name because of using SHA (commit hash)
 
Please fix it

**Comment by Vasiliy-Lavrov** [comment]
I apply a patch for oao v1.2.1 sources in the Jenkinsfile:

```groovy 
script {
    writeFile file: "oao_patch.diff", text: '''
--- publish.js	2018-02-06 12:34:28.041469452 +0300
+++ publish_2.js	2018-02-06 13:08:19.140511452 +0300
@@ -91,8 +91,8 @@
             return _context.abrupt('return');
 
           case 10:
-            _context.next = 12;
-            return prepublishChecks({ master: master, checkUncommitted: checkUncommitted, checkUnpulled: checkUnpulled });
+            // _context.next = 12;
+            // return prepublishChecks({ master: master, checkUncommitted: checkUncommitted, checkUnpulled: checkUnpulled });
 
           case 12:
             _context.next = 14;
'''

    sh """
        patch node_modules/oao/lib/publish.js < oao_patch.diff
    """
}
```

It works for me.

**Comment by guigrpa** [comment]
Have you checked with `oao publish -i "examples/*" --no-master --no-changelog --no-git-commit --no-check-uncommitted --no-confirm --new-version 0.1.0-0205151508.alpha --no-check-unpulled`? Notice the last flag.

**Comment by guigrpa** [comment]
I've seen where your problem is. Even if you by-pass checks (e.g. with `no-check-uncommitted`, oao asks git for the information in order to *warn* you). I've added a `--no-checks` flag to `oao publish` which bypasses the whole prepublishing checks. Shipped in **v1.3.1**.

**Comment by Vasiliy-Lavrov** [comment]
`oao publish -i \"examples/*\" --no-checks --no-git-commit --no-changelog --no-confirm --new-version 0.1.0-213153115.alpha`
It works! Thx!
